### ISSUE TITLE
Question about NLLLoss

### ISSUE THREAD

**Issue by kaichengyang0828** [jamesli1618/Obj-GAN#30]
In the bbox generation model, you use NLLLoss to comput the loss of label prediction, but before use NLLLoss you just use softmax function without torch.log, so the value of lloss is always a negative number. I don't understand what the purpose of this operation is.

**Comment by sbkim052** [comment]
@Kaicheng-Yang0828
Yeah. I agree. Did you fix that error and successfully train it on your setting?

**Comment by fortunechen** [comment]
@Kaicheng-Yang0828 @sbkim052 
Do you reproduce the result? I can not reproduce the bbox generation according to the default setting :(

**Comment by fortunechen** [comment]
I can reproduce the result  by making the following changes:

1. set lr=0.0001 according to https://github.com/jamesli1618/Obj-GAN/issues/27#issue-771979855
2. set lamda2=0.25 according to https://dl.acm.org/doi/10.1145/3474085.3475363
3. don't use label reweight loss by comment https://github.com/jamesli1618/Obj-GAN/blob/44b442ff1a1588bcf7c4855aaca2e6a4c26b10ed/box_generation/sample.py#L98-https://github.com/jamesli1618/Obj-GAN/blob/44b442ff1a1588bcf7c4855aaca2e6a4c26b10ed/box_generation/sample.py#L103 
4. correct the label loss from -P to -logP as you pointed out

**Comment by xiaozhamao** [comment]
> I can reproduce the result by making the following changes:
> 
> 1. set lr=0.0001 according to [NaN  error when training the box_generation model #27 (comment)](https://github.com/jamesli1618/Obj-GAN/issues/27#issue-771979855)
> 2. set lamda2=0.25 according to https://dl.acm.org/doi/10.1145/3474085.3475363
> 3. don't use label reweight loss by comment https://github.com/jamesli1618/Obj-GAN/blob/44b442ff1a1588bcf7c4855aaca2e6a4c26b10ed/box_generation/sample.py#L98-https://github.com/jamesli1618/Obj-GAN/blob/44b442ff1a1588bcf7c4855aaca2e6a4c26b10ed/box_generation/sample.py#L103
> 4. correct the label loss from -P to -logP as you pointed out

I am confused about the change 3. Did you just delete the code and use initialization of the weight.
Hope to get your reply as soon as possible. 
Thx.

**Comment by fortunechen** [comment]
yes
